metaflow/metadata/heartbeat.py

```diff
@@ -20,27 +20,30 @@ def __init__(self, msg):
 class MetadataHeartBeat(object):
     def __init__(self):
         self.headers = METADATA_SERVICE_HEADERS
-        self.req_thread = Thread(target=self.ping)
+        self.req_thread = Thread(target=self._ping)
         self.req_thread.daemon = True
         self.default_frequency_secs = 10
         self.hb_url = None
 
     def process_message(self, msg):
         # type: (Message) -> None
         if msg.msg_type == MessageTypes.SHUTDOWN:
-            # todo shutdown doesnt do anything yet? should it still be called
-            self.shutdown()
-        if (not self.req_thread.is_alive()) and msg.msg_type == MessageTypes.LOG_EVENT:
+            self._shutdown()
+        if (not self.req_thread.is_alive()) and msg.msg_type == MessageTypes.MUST_SEND:
```

Comment:
The heartbeats are best effort. `BEST_EFFORT_SEND`?

Reply:
this is the heartbeat configuration. It doesn't really make a difference though so I can change it. Semantically I thought that the configuration was more of a MUST_SEND but since there are no other messages sent anyways, it has no impact.

Reply:
in that case, do you need to check for the message type here?

Reply:
not really. I'll remove.